Repository/PhpcrOdmRepository.php

```diff
@@ -116,4 +119,20 @@ protected function buildCollection(array $documents)
 
         return $collection;
     }
+
+    /**
+     * Returns all versions of a resource.
+     *
+     * @param string $path The path to the resource.
+     *
+     * @return VersionList The versions stored for this path.
+     *
+     * @throws NoVersionFoundException  If no version can be found.
+     * @throws InvalidArgumentException If the path is invalid. The path must
+     *                                  be a non-empty string starting with "/".
+     */
+    public function getVersions($path)
+    {
+        return new VersionList($path, []);
```

Comment:
Yeah, a getter test is fine. It will ensure that the __construct
arguments work and that the class is correctly included.

On Thu, May 19, 2016 at 04:37:13AM -0700, Maximilian Berghoff wrote:
>    In [1]Repository/PhpcrOdmRepository.php:
> 
>  > @@ -116,4 +119,20 @@ protected function buildCollection(array $documents)
>  >
>  >          return $collection;
>  >      }
>  > +
>  > +    /**
>  > +     * Returns all versions of a resource.
>  > +     *
>  > +     * @param string $path The path to the resource.
>  > +     *
>  > +     * @return VersionList The versions stored for this path.
>  > +     *
>  > +     * @throws NoVersionFoundException  If no version can be found.
>  > +     * @throws InvalidArgumentException If the path is invalid. The path must
>  > +     *                                  be a non-empty string starting with "/".
>  > +     */
>  > +    public function getVersions($path)
>  > +    {
>  > +        return new VersionList($path, []);
> 
>    it was just a quick fix to implement it, no logic. So the test would be
>    like a setter-getter test ;-) In generall we should add the phpcr
>    versioning logic when available. Am 19.05.2016 um 13:28 schrieb dantleech:
>    [2]…
>    In Repository/PhpcrOdmRepository.php <[3]#15 (comment)>: > @@ -116,4
>    +119,20 @@ protected function buildCollection(array $documents) > > return
>    $collection; > } > + > + /** > + * Returns all versions of a resource. > +
>    * > + * [4]@param string $path The path to the resource. > + * > + *
>    [5]@return VersionList The versions stored for this path. > + * > + *
>    [6]@throws NoVersionFoundException If no version can be found. > + *
>    [7]@throws InvalidArgumentException If the path is invalid. The path must
>    > + * be a non-empty string starting with "/". > + */ > + public function
>    getVersions($path) > + { > + return new VersionList($path, []); Can you
>    add a test for this in |PhpcrOdmRepositoryTest| ? — You are receiving this
>    because you authored the thread. Reply to this email directly or view it
>    on GitHub
>    <[8]https://github.com/symfony-cmf/resource/pull/15/files/bb855320dfc8f8e54706c37dea6693feb52ce691#r63861958>
> 
>    —
>    You are receiving this because you commented.
>    Reply to this email directly or [9]view it on GitHub
> 
>    Reverse link: [10]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/symfony-cmf/resource/pull/15#discussion_r63862895
>    2. file:///dev/stdin#
>    3. https://github.com/symfony-cmf/resource/pull/15#discussion_r63861958
>    4. https://github.com/param
>    5. https://github.com/return
>    6. https://github.com/throws
>    7. https://github.com/throws
>    8. https://github.com/symfony-cmf/resource/pull/15/files/bb855320dfc8f8e54706c37dea6693feb52ce691#r63861958
>    9. https://github.com/symfony-cmf/resource/pull/15/files/bb855320dfc8f8e54706c37dea6693feb52ce691#r63862895
>   10. https://github.com/symfony-cmf/resource/pull/15/files/bb855320dfc8f8e54706c37dea6693feb52ce691#r63862895